KubeArmor/core/k8sHandler.go

```diff
@@ -209,38 +210,49 @@ func (kh *K8sHandler) DoRequest(cmd string, data interface{}, path string) ([]by
 // ========== //
 
 // WatchK8sNodes Function
-func (kh *K8sHandler) WatchK8sNodes() *http.Response {
+func (kh *K8sHandler) WatchK8sNodes() watch.Interface {
 	if !kl.IsK8sEnv() { // not Kubernetes
 		return nil
 	}
 
 	if kl.IsInK8sCluster() { // kube-apiserver
-		URL := "https://" + kh.K8sHost + ":" + kh.K8sPort + "/api/v1/nodes?watch=true"
 
-		req, err := http.NewRequest("GET", URL, nil)
-		if err != nil {
-			return nil
-		}
-
-		req.Header.Add("Content-Type", "application/json")
-		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", kh.K8sToken))
+		if kh.InitInclusterAPIClient() {
+			watchNodes, err := kh.K8sClient.
+				CoreV1().
+				Nodes().
+				Watch(
+					context.Background(),
+					metav1.ListOptions{},
+				)
+			defer watchNodes.Stop()
```

Comment:
Wouldn't this stop the watcher after returning it?
In the test scenario, KubeArmor is unable to get node information which means this is not working as expected I believe 🤔

Reply:
Ah, I see, `defer` has called twice which overrides the watcher. Thanks @daemon1024 for helping me fix. 😄 ✌🏻